src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -483,15 +483,20 @@ private void StartConfiguring()
         }
 
         private void ThrowForNonFreeStatus(int status)
+        {
+            throw GetExceptionForNonFreeStatus(status);
+        }
+
+        private Exception GetExceptionForNonFreeStatus(int status)
```

Comment:
If this does end up being beneficial for call sites of ThrowForNonFreeStatus, can this Get method be made into a local?

Reply:
Good idea